### ISSUE TITLE
Need help in passing the config yaml as a parameter

### ISSUE THREAD

**Issue by dpanyam** [peer-calls/peer-calls#152]
Hi ,

 I see this from the README  "**_Only a single ICE server can be defined via environment variables. To define more use a YAML config file. To load a config file, use the -c /path/to/config.yml command line argument_**"

My question is for the above to work do i need to rebuild the binary.  I want a self contained docker image that uses config yml and i don't want to pass the certs as command line arguments. In my docker file i changed the entrypoint to something like this `_ENTRYPOINT ["/usr/local/bin/peer-calls","-c","/usr/local/bin/config.yml"]_` is this correct?

**Comment by chrismgonzalez** [comment]
This issue has been solved.  @dpanyam It would be great if we could write up an addendum to the docs and get a PR in so we can have that documentation here for others.

**Comment by dpanyam** [comment]
Below are the changes in code / docker build to  have more than 1 ICE servers

### Problem Statement ###
The default setup allows only 1 ICE server to be passed in as command line arguments. To support multiple ICE servers a config.yml file needs to be created

### Steps ###

1. Edit config.yml  in peer-calls/server/config.yml  to add the STUN/TURN servers for your setup
2. Add a COPY step in Dockerfile to copy over the config.yml  file to the image
3. Modify the ENTRYPOINT in Dockerfile to pass this as command line argument. For example i copied this file over to /usr/local/bin so my ENTRYPOINT looks like ENTRYPOINT ["/usr/local/bin/peer-calls","-c","/usr/local/bin/config.yml"]
4. Build the docker image

**Comment by jeremija** [comment]
I'm late to the party but I'm pretty sure you can specify an external config file without rebuilding the image.

For example:

```
docker run -v /path/to/config.yml:/config.yml peercalls/peercalls:latest -c /config.yml
```

*not sure if the above command works since I typed it on my mobile device, but you should get the picture.

**Comment by dpanyam** [comment]
Okay. I am not sure if the position of that parameter matters but i see -c option with docker run command used to specify the CPU shares

**Comment by jeremija** [comment]
Any arguments after specified after image name will be passed to whatever is defined in image's `ENTRYPOINT`, which is the `peer-calls` binary in this case:

```
docker run [OPTIONS] IMAGE [COMMAND] [ARG...]
```


Source: https://docs.docker.com/engine/reference/commandline/run/

**Comment by dpanyam** [comment]
Thanks yes agree.

**Comment by harikt** [comment]
I actually tried to run passing with -c option when starting docker container.

`docker run --name peer-calls -d --rm -it -p 3000:3000 -v /local/peercalls/config.yml:/peercalls/config.yml peer-calls -c /peercalls/config.yml`

But the container itself is not starting.
Without passing the config, I tried to log in to container via 

```
docker exec -it peer-calls sh
docker exec -it peer-calls bash
docker exec -it peer-calls /bin/bash
docker exec -it peer-calls /bin/ash
docker exec -it peer-calls /bin/sh
```

For all the above commands it was throwing

```
OCI runtime exec failed: exec failed: container_linux.go:380: starting container process caused: exec: "sh": executable file not found in $PATH: unknown
```

Looking at https://stackoverflow.com/a/44804509/487878, it suggests

> Install Bash in your Alpine image, as you seem to expect Bash is present, with such a line in your Dockerfile:

RUN apk add --no-cache --upgrade bash

Any thoughts ?

**Comment by prologic** [comment]
Well first of all your image name was wrong (_or you built your own_):

- The image is called `peercalls/peercalls`

Secondly with the `--rm` option you would never be able to find out why it didn't start because the container would have been removed instantly on failure.

Removing the `--rm` option I see this failure:

```
152f244e8993   peercalls/peercalls         "/usr/local/bin/peer…"   3 seconds ago   Exited (1) 3 seconds ago
```

**Comment by prologic** [comment]
If I run interactively with a sample config I pulled from the [README](/README.md) I can see clearly why it wouldn't start:

```#!sh
$ docker run -i -t -p 3000:3000 -v $PWD/config.yml:/peercalls/config.yml peercalls/peercalls -c /peercalls/config.yml
2021-08-07T00:46:07.133284Z error [                main] Command error: yaml: line 7: did not find expected key
/src/server/configread.go:59: decode yaml
/src/server/configread.go:23: read yaml config: /peercalls/config.yml
/src/server/configread.go:30:
/src/server/configread.go:53:
/src/server/cli/server.go:89: read config
/src/server/cli/server.go:37:
/src/server/command/command.go:218:
/src/server/command/command.go:239:
/src/server/cli/cli.go:22:
/src/main.go:55:
/src/main.go:81:
```

Perhaps you have a similar problem?

**Comment by prologic** [comment]
As to your final problem, you cannot "exec" anything inside the container period. This is because the runtime layer is using the `scratch` image:

https://github.com/peer-calls/peer-calls/blob/0a9ac6afe8406eb8c3d36962581b7fd0af528f0b/Dockerfile#L45-L52

This is pretty typical and normal for many Go applications or things that doesn't require any userland. I would suggest you debug your usage as per my comments above.

**Comment by prologic** [comment]
Recommend closing and locking this issue now 😀

**Comment by harikt** [comment]
Hey @prologic ,

The information you shared is correct and thank you.

1 . I am building the container.
2. Thanks for the tip regarding --rm option
3. The problem was with the config.

But even if I pass the correct config.yml file, I noticed in the `index.js` that the stun server used is the default ones.

```
p.config={iceServers:[{urls:["stun:stun.l.google.com:19302","stun:global.stun.twilio.com:3478"]}],
```

What I passed in the config.yml was

```
icse_servers:
- urls:
  - 'stun:coturn.buildful.com:5349'
```

I can open another issue if that is a different one.
